src/mappings/entities/mapPortfolio.ts

```diff
@@ -0,0 +1,146 @@
+  serializeTicker,
+} from '../util';
+import { EventIdEnum, ModuleIdEnum } from './common';
+import { createLeg, SettlementResultEnum } from './mapSettlement';
+
+export const getPortfolio = async ({
+  identityId,
+  number,
+}: Pick<Portfolio, 'identityId' | 'number'>): Promise<Portfolio> => {
+  const portfolioId = `${identityId}/${number}`;
+
+  let portfolio = await Portfolio.get(portfolioId);
+
+  if (!portfolio) {
+    if (number === 0) {
+      portfolio = Portfolio.create({
+        id: portfolioId,
+        identityId,
+        number,
+        name: 'Default Portfolio',
+      });
+      await portfolio.save();
+    } else {
+      throw new Error(`Portfolio number ${number} not found for DID ${identityId}`);
+    }
+  }
+
+  return portfolio;
+};
+
+const handlePortfolioCreated = async (blockId: string, params: Codec[]): Promise<void> => {
+  const [rawOwnerDid, rawPortfolioNumber, rawName] = params;
+
+  const ownerId = getTextValue(rawOwnerDid);
+  const number = getNumberValue(rawPortfolioNumber);
+  const name = getTextValue(rawName);
+
+  await Portfolio.create({
+    id: `${ownerId}/${number}`,
+    identityId: ownerId,
+    number,
+    name,
+    createdBlockId: blockId,
+  }).save();
+};
+
+const handlePortfolioRenamed = async (blockId: string, params: Codec[]): Promise<void> => {
+  const [rawOwnerDid, rawPortfolioNumber, rawName] = params;
+
+  const ownerId = getTextValue(rawOwnerDid);
+  const number = getNumberValue(rawPortfolioNumber);
+  const name = getTextValue(rawName);
+
+  const portfolio = await getPortfolio({ identityId: ownerId, number });
+  portfolio.name = name;
+  portfolio.updatedBlockId = blockId;
+
+  await portfolio.save();
+};
+
+const handlePortfolioDeleted = async (params: Codec[]): Promise<void> => {
+  const [rawOwnerDid, rawPortfolioNumber] = params;
+
+  const ownerId = getTextValue(rawOwnerDid);
+  const number = getNumberValue(rawPortfolioNumber);
+
+  await Portfolio.remove(`${ownerId}/${number}`);
+};
+
+const handlePortfolioCustodianChanged = async (blockId: string, params: Codec[]): Promise<void> => {
+  const [, rawPortfolio, rawCustodian] = params;
+
+  const portfolioValue = getPortfolioValue(rawPortfolio);
+  const custodian = getTextValue(rawCustodian);
+
+  const portfolio = await getPortfolio(portfolioValue);
+  portfolio.custodianId = custodian;
+  portfolio.updatedBlockId = blockId;
+
+  await portfolio.save();
+};
+
+const handlePortfolioMovement = async (
+  blockId: string,
+  eventId: EventIdEnum,
+  params: Codec[],
+  event: SubstrateEvent
+): Promise<void> => {
+  const [, rawFromPortfolio, rawToPortfolio, rawTicker, rawAmount] = params;
+  const address = getSignerAddress(event);
+  const from = getPortfolioValue(rawFromPortfolio);
+  const to = getPortfolioValue(rawToPortfolio);
+  const ticker = serializeTicker(rawTicker);
+  const amount = getBigIntValue(rawAmount);
+
+  const settlementId = `${blockId}/${event.idx}`;
+  const settlement = Settlement.create({
```

Comment:
I thought a portfolio movement was different from a Settlement. Are you sure we need to make a settlement for this?

It looks like this is how it was before, but since we are changing things I am thinking we'd be able to split this up now. Seems a bit messy shoe horning it into Settlements.

I might be wrong and these actually are Settlements, but I am thinking they are different things on chain.

Reply:
I kept it as it is because in tooling while fetching settlements data `MovedBetweenPortfolios` was considered. I'm not sure we would want this, but as current use case it fetches all settlements where a DID or a specific portfolio is a part of and then also considers movement between portfolios. Ref -> https://github.com/PolymathNetwork/tooling-gql/blob/develop/src/controllers/settlement.ts

Reply:
@VictorVicente Would like to know your thoughts here

Reply:
I agree we should avoid to mix them here. We are forcing portfolio movements to fit into settlement model when they are not the same. 
I think we should track these movements in a different entity.
And the SDK should mix them when getting `portfolioTransactionHistory`. SDK could fetch them with two different calls and merge them into the same DTO by ordering them by block id to keep the same results we have right now.

WDYT @polymath-eric @prashantasdeveloper ?

Reply:
+1 to mixing them in the SDK. The reason they are like that is because that was what the dashboard wanted, so the SDK seems like a good place to handle that concern.

Reply:
Dashboard requirement makes sense. They want to show the complete history for a portfolio. That includes portfolio movements, without them balances could not be ok in that history. 
But it makes sense to track them independently in SubQuery, and return the whole history in the SDK IMO.